File: projet/source/f1levier/app/src/main/java/fr/utbm/lo52/f1levier/db/dao/ParticipantDao.java
package fr.utbm.lo52.f1levier.db.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

import fr.utbm.lo52.f1levier.db.entity.Participant;

@Dao
public interface ParticipantDao {
    @Query("SELECT * FROM Participant ORDER BY name")
    LiveData<List<Participant>> getAll();

    @Query("SELECT * FROM Participant WHERE id IN (:ids) ORDER BY name")
    List<Participant> getAllByIds(final int[] ids);

    @Query("SELECT * FROM Participant WHERE id = :id")
    Participant getById(final int id);

    @Insert
    void insert(Participant participant);
}

File: CourseCODEP25/app/src/main/java/com/utbm/nouassi/manou/coursecodep25/database/dao/ParticipantDao.kt
package com.utbm.nouassi.manou.coursecodep25.database.dao

import androidx.room.*
import com.utbm.nouassi.manou.coursecodep25.database.entity.Participant

@Dao
interface ParticipantDao {
    @Insert
    fun insertAll(vararg participants: Participant)

    @Insert
    fun insert(participant: Participant) : Long

    @Delete
    fun delete(participant: Int)

    @Query("SELECT * FROM participant ORDER bY id desc")
    fun getAll(): List<Participant>

    @Update
    fun updateParticipants(vararg participants: Participant)

    @Query("SELECT * FROM participant where id = :id")
    fun getById(id : Int): Participant

    @Query("SELECT * FROM participant order by niveau desc")
    fun getAllOrderNiveauDesc(): List<Participant>
}